### PR TITLE
Add Synchronous to MySqlConnectionStringBuilder.

### PR DESCRIPTION
**Pull request by ejball**
This initial implementation could have adverse affects on performance, so you might not want to merge, but it is an idea for enforcing synchronous access, even when using the async methods, to improve debugging.

### CODE REVIEW

File: src/MySqlConnector/MySqlClient/MySqlConnection.cs [hunk 1]
```diff
@@ -240,6 +240,32 @@ internal MySqlSession Session
...
+
+			return Task.FromResult(task.GetAwaiter().GetResult());
+		}
+
+		internal ValueTask<T> AdaptTask<T>(ValueTask<T> task)
+		{
+			if (!Synchronous)
+				return task;
+
+			return new ValueTask<T>(task.AsTask().GetAwaiter().GetResult());
```
Comment (bgrainger) [inline src/MySqlConnector/MySqlClient/MySqlConnection.cs]: Should this have an optimised code path for the case where the `ValueTask` already has a value (which is expected to be common, hence the use of `ValueTask`)?

Comment (bgrainger) [inline src/MySqlConnector/MySqlClient/MySqlConnection.cs]: Maybe `.AsTask()` returns a completed task efficiently; but still there is unnecessary allocation.

**Comment by bgrainger** [comment]
I'm not convinced I like this approach. I haven't attempted it, but it feels like one could make the socket operations synchronous, then use `ValueTask` to bubble the results as far up the call stack as possible. (Then, the remaining uses of `Task` would still be synchronous.)

This ties into an improvement I want to make to use all new non-public methods that return `ValueTask` in ExecuteDbDataReaderAsync and reimplement the public API in terms of those new ValueTask methods.

File: src/MySqlConnector/MySqlClient/MySqlConnection.cs (hunk 1 above)
Comment (ejball) [inline src/MySqlConnector/MySqlClient/MySqlConnection.cs]: Yes, that would be a great change.

**Comment by ejball** [comment]
I don't like it either, but I wasn't ready to do the hard work of using synchronous socket operations. This is just a proof of concept that may encourage someone to eventually do it right. :-)

**Comment by ejball** [comment]
Rebased.

**Comment by bgrainger** [comment]
> wasn't ready to do the hard work of using synchronous socket operations

Turned out to not be that hard; code is referenced from #62. I'm going to close this PR in favour of (the eventual resolution of) that one.

**Pull request closed without merging**